scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java

```diff
@@ -1162,47 +1155,81 @@ public boolean equals(Object obj) {
         return false;
     }
 
+    public synchronized void updateVariables(SchedulingService schedulingService) throws IOException, ClassNotFoundException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.lang.ClassNotFoundException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1160)